R/cBioDataPack.R

```diff
@@ -91,28 +91,30 @@ cBioDataPack <- function(cancer_study_id, use_cache = TRUE,
     filelist <- gsub("^\\.\\/", "", filelist)
     filekeepind <- grep("^\\._", basename(filelist), invert = TRUE)
     filelist <- filelist[filekeepind]
-    ## Remove files that are corrupt / hidden (start with ._)
-    datafiles <- grep(x = filelist, pattern = "data.*\\.(txt|seg)$",
-        value = TRUE)
-    datafiles <- c(datafiles, grep("meta_study", filelist, value = TRUE),
-        grep("/LICENSE", filelist, value = TRUE))
+    datafiles <- getRelevantFilesFromStudy(filelist)
 
     worktemp <- tempdir()
     untar(cancer_file, files = datafiles, exdir = worktemp,
         extras = exarg)
+    extractedStudyFolder2MultiAssayExperiment(worktemp, names.field)
+}
+
+extractedStudyFolder2MultiAssayExperiment <- function(filepath,
```

Comment:
This function should start higher up so that it can seamlessly connect to `downloadStudy`.